packages/jupyter-ai/jupyter_ai/extension.py

```diff
@@ -172,6 +181,8 @@ def initialize_settings(self):
         generate_chat_handler = GenerateChatHandler(
             **chat_handler_kwargs,
             root_dir=self.serverapp.root_dir,
+            preferred_dir=self.serverapp.contents_manager.preferred_dir,
```

Comment:
Note to self/reviewers: there is also `serverapp.preferred_dir` but it is deprecated. The difference is that `serverapp.contents_manager.preferred_dir` can be an empty string while `serverapp.preferred_dir` had a built-in fallback to `root_dir`.